src/v/cluster/cloud_metadata/cluster_manifest.h

```diff
@@ -68,6 +69,8 @@ struct cluster_metadata_manifest
     // changed since the last snapshot.
     ss::sstring controller_snapshot_path;
 
+    std::vector<std::vector<ss::sstring>> offsets_snapshots_by_partition;
```

Comment:
i'm on the fence with this vector of vector, i would prefer a more expressive `[whatever]_map<partition_id, vector<remote_path>>` but it might be overkill and not compatible with the wire format

Reply:
Yeah tbh I did it this way to avoid having a map in the json, which would I think is a bit annoying